### PR TITLE
Add ability to delete a task

### PR DESCRIPTION
**Pull request by ajvb**
delete

### TODO's

- [ ] Disable the "delete" icon on all non-leaf nodes.

### CODE REVIEW

File: src/api/postgrest.js [hunk 1]
```diff
@@ -4,18 +4,23 @@ export const formatDate = (rfc3339, displayFormat="YYYY-MM-DD") => {
...
-export const postJSON = (pathSuffix, payload, additionalHeaders = {}) => {
-  const headers = {
-    'Content-Type': 'application/json; charset=utf-8'
-  }
+const defaultHeaders = {
+  'Content-Type': 'application/json; charset=utf-8'
+}
+
+const createHeaders = (additionalHeaders) => {
+  const headers = {}
```
Comment (elimisteve) [inline src/api/postgrest.js]: :cry:

**Comment by ajvb** [comment]
Currently getting the error `TypeError: _this.props.deleteTask is not a function` on child tasks.

File: src/components/Content/TaskHierarchy/Task/Task.js [hunk 2]
```diff
@@ -38,6 +40,22 @@ class Task extends Component {
...
+  deleteTaskBtn = () => {
+    //TODO: Consider adding a confirmation box
+    if (this.props.taskData.subtask_gids.length !== 0) {
+      return;
+    }
+    this.props.deleteTask(this.props.taskData.gid);
+  }
+
+  deleteTaskBtnClasses = () => {
+    var btnClasses = "icon-ctn";
```
Comment (elimisteve) [inline src/components/Content/TaskHierarchy/Task/Task.js:52]: @ajvb FYI, `let` is the new `var`

**Comment by elimisteve** [comment]
Updated version of this added in https://github.com/PursuanceProject/pursuance/commit/ba718e63887594277c3cabdc3f920dabdf810f9f by @colinfruit! :tada: 

Thank you both :+1:

**Pull request closed without merging**